### PR TITLE
TS-19-v2-23: Refactor ESLint errors in components and helper directories

### PR DESCRIPTION
**Pull request by Akr1s**

### CODE REVIEW

File: frontend/src/components/ChangePasswordForm/ChangePasswordForm.js [hunk 1]
```diff
@@ -94,7 +92,7 @@ const ChangePasswordForm = (props) => {
     );
 };
 
-const mapStateToProps = (state) => ({});
+const mapStateToProps = () => ({});
```
Comment (HiroFive) [inline frontend/src/components/ChangePasswordForm/ChangePasswordForm.js]: Why should we leave mapStateToProps if it doesn't return anything anyway

File: frontend/src/components/Navigation/NavigationPage.js [hunk 2]
```diff
@@ -99,15 +73,15 @@ const NavigationPage = (props) => {
                     scrollButtons="on"
                     className={classes.header}
                 >
-                    {tabsComponents.map((tab_one, index) => (
+                    {tabsComponents.map((tabOne, index) => (
                         <>
-                            {tab_one.length === undefined ? (
-                                <Link className={classes.nav} to={links[tab_one.name]}>
+                            {tabOne.length === undefined ? (
```
Comment (HiroFive) [inline frontend/src/components/Navigation/NavigationPage.js]: Replace with the isNil() function with lodash

File: frontend/src/components/Navigation/NavigationPage.js [hunk 3]
```diff
@@ -1,43 +1,17 @@
...
-    );
-}
-
-TabPanel.propTypes = {
-    children: PropTypes.node,
-    index: PropTypes.any.isRequired,
-    value: PropTypes.any.isRequired,
-};
-
 function a11yProps(index) {
```
Comment (HiroFive) [inline frontend/src/components/Navigation/NavigationPage.js:16]: Replace the strange function name

File: frontend/src/components/TemporarySchedule/TemporaryScheduleCardButtons/TemporaryScheduleCardButtons.js [hunk 4]
```diff
@@ -34,12 +35,13 @@ const TemporaryScheduleCardButtons = (props) => {
...
-                              ...schedule,
-                              date,
-                          });
+                    if (isTemporary) selectTemporarySchedule({ ...schedule, date });
+                    else {
+                        handleScheduleSelect({
+                            ...schedule,
+                            date,
+                        });
+                    }
```
Comment (HiroFive) [inline frontend/src/components/TemporarySchedule/TemporaryScheduleCardButtons/TemporaryScheduleCardButtons.js]: It is better to make it a separate function

File: frontend/src/helper/disableComponent.js [hunk 5]
```diff
@@ -1,16 +1,8 @@
...
-        return false;
-    }
-    if (!pristine) {
-        return false;
-    }
-    if (submitting) {
-        return true;
-    }
-    if (id === undefined) {
+    if (submitting || id === undefined) {
```
Comment (HiroFive) [inline frontend/src/helper/disableComponent.js]: It would be better to replace it with the isNil function with lodash

**Review by HiroFive — APPROVED** [review]

File: frontend/src/components/Board/Board.js [hunk 6]
```diff
@@ -80,8 +82,8 @@ const Board = (props) => {
     const addClassDayBoard = () => {
         if (classDay !== null && classDay !== undefined) {
             const dayClassWeek = document.getElementsByClassName(classDay);
-            dayClassWeek[0].classList.add('day-class-week');
-            dayClassWeek[1].classList.add('day-class-week');
+            dayClassWeek[0].classList.add(dayClassWeekString);
+            dayClassWeek[1].classList.add(dayClassWeekString);
         }
     };
     const addClass = () => {
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/Board/Board.js:89]: Could you use isNill inside the function?

File: frontend/src/share/renderedFields/timeSemester.js [hunk 7]
```diff
@@ -14,7 +15,7 @@ const renderMounthPicker = ({
             <DatePicker
                 minDate={new Date()}
                 clearable
-                value={value ? moment(value, 'DD/MM/YYYY').toDate() : null}
+                value={value ? moment(value, dateFormat).toDate() : null}
```
Comment (ViktoriiaKosovan) [inline frontend/src/share/renderedFields/timeSemester.js:18]: You can use && operator

**Review by ViktoriiaKosovan — APPROVED** [review]

File: frontend/src/components/CopyLessonsFromSemesterForm/CopyLessonsFromSemesterForm.js [hunk 8]
```diff
@@ -13,37 +13,34 @@ import Card from '../../share/Card/Card';
...
+                    ))}
+                </Field>
+            );
+        }
+        if (semesters && semesters.length === 1) {
+            handleSubmit({ fromSemesterId: semesters[0].id });
+            return <p>{semesters[0].description}</p>;
         }
+        return null;
     };
```
Comment (andrii-kuzmichov) [inline frontend/src/components/CopyLessonsFromSemesterForm/CopyLessonsFromSemesterForm.js]: Use swtich

File: frontend/src/components/FreeRoomForm/freeRoomForm.js [hunk 9]
```diff
@@ -17,14 +17,12 @@ let FreeRoomForm = (props) => {
...
+    const { handleSubmit, classScheduler, pristine, submitting, onReset, semesters } = props;
 
-    const class_names = [];
+    const classNames = [];
 
     if (classScheduler.length - 1 > 0) {
-        classScheduler.map((classSchedulerOne) => {
-            class_names.push(classSchedulerOne.class_name);
-        });
+        classScheduler.map((classSchedulerOne) => classNames.push(classSchedulerOne.class_name));
```
Comment (andrii-kuzmichov) [inline frontend/src/components/FreeRoomForm/freeRoomForm.js:25]: incorrect, let use forEach

File: frontend/src/components/GroupSchedulePage/GroupSchedulePage.js [hunk 10]
```diff
@@ -55,26 +46,23 @@ const GroupSchedulePage = (props) => {
...
-    const emptySchedule = () => <p className="empty_schedule">{t('common:empty_schedule')}</p>;
     const { t } = useTranslation('common');
-    const renderDownloadLink = (entity, semesterId, entityId) => {
+    const emptySchedule = () => <p className="empty_schedule">{t('common:empty_schedule')}</p>;
+    const renderDownloadLink = (entity, semester, entityId) => {
         let link = '';
         const { language } = i18n;
         const languageToRequest = `&language=${language}`;
-        if (semesterId && entityId) {
+        if (semester && entityId) {
```
Comment (andrii-kuzmichov) [inline frontend/src/components/GroupSchedulePage/GroupSchedulePage.js:59]: why you rename it?

File: frontend/src/components/GroupSchedulePage/GroupSchedulePage.js [hunk 11]
```diff
@@ -118,8 +107,8 @@ const GroupSchedulePage = (props) => {
     };
 
     const renderSchedule = () => {
-        switch (props.scheduleType) {
-            case 'group':
+        switch (scheduleType) {
+            case 'group': {
                 if (
                     (!groupSchedule ||
                         (groupSchedule.schedule && groupSchedule.schedule.length === 0)) &&
```
Comment (andrii-kuzmichov) [inline frontend/src/components/GroupSchedulePage/GroupSchedulePage.js:114]: use `isEmpty` from lodash

File: frontend/src/components/GroupSchedulePage/GroupSchedulePage.js [hunk 12]
```diff
@@ -191,7 +181,7 @@ const GroupSchedulePage = (props) => {
                     }
                 } else setLoadingService(false);
                 break;
-            case 'full':
+            case 'full': {
                 if (
                     (!fullSchedule.schedule || fullSchedule.schedule.length === 0) &&
```
Comment (andrii-kuzmichov) [inline frontend/src/components/GroupSchedulePage/GroupSchedulePage.js:186]: the same as above

File: frontend/src/components/GroupSchedulePageTop/GroupSchedulePageTop.js [hunk 13]
```diff
@@ -56,39 +54,38 @@ const GroupSchedulePageTop = (props) => {
...
+                        <option key={semester.id} value={semester.id}>
+                            {semester.description}
+                        </option>
+                    ))}
+                </Field>
+            );
+        }
+        if (semesters && semesters.length === 1) {
+            handleSubmit({ semester: semesters[0].id });
+            return <p>{semesters[0].description}</p>;
```
Comment (andrii-kuzmichov) [inline frontend/src/components/GroupSchedulePageTop/GroupSchedulePageTop.js:86]: switch and maybe reuse above functionality

File: frontend/src/components/LessonForm/LessonForm.js [hunk 14]
```diff
@@ -65,30 +82,13 @@ let LessonForm = (props) => {
...
-            {groupId ? (
+            {groupId && (
                 <h2 className="form-title under-line">
                     {lessonId ? t('edit_title') : t('create_title')}
                     {t('lesson_label')}
                 </h2>
-            ) : (
-                ''
             )}
             {groupId ? (
```
Comment (andrii-kuzmichov) [inline frontend/src/components/LessonForm/LessonForm.js:93]: refactor condition for `groupId`

File: frontend/src/components/LessonsList/LessonsList.js [hunk 15]
```diff
@@ -36,6 +36,7 @@ const LessonsList = (props) => {
         if ((number >= 5 && number <= 20) || number === 0) {
             return 'годин';
         }
+        return 'години';
```
Comment (andrii-kuzmichov) [inline frontend/src/components/LessonsList/LessonsList.js:39]: to constants

File: frontend/src/components/LessonsList/LessonsList.js [hunk 16]
```diff
@@ -60,14 +61,10 @@ const LessonsList = (props) => {
     };
     const getHour = (number) => {
         const language = i18n.language.toUpperCase();
-        const en = 'EN';
-        const uk = 'UK';
-        if (language === en) {
+        if (language === 'EN') {
```
Comment (andrii-kuzmichov) [inline frontend/src/components/LessonsList/LessonsList.js:64]: as above

File: frontend/src/components/Navigation/NavigationPage.js [hunk 17]
```diff
@@ -118,18 +93,18 @@ const NavigationPage = (props) => {
                                     id="demo-controlled-open-select"
                                     value={gen}
                                     onChange={(event) => {
-                                        const val = event.target.value;
-                                        setGen(val);
-                                        document_title(val);
+                                        const { eventValue } = event.target;
+                                        setGen(eventValue);
+                                        documentTitle(eventValue);
```
Comment (andrii-kuzmichov) [inline frontend/src/components/Navigation/NavigationPage.js:98]: are you sure that this property exists in the target?

File: frontend/src/components/ScheduleDialog/ScheduleDialog.js [hunk 18]
```diff
@@ -22,6 +22,19 @@ const useStyles = makeStyles(() => ({
     },
 }));
 
+const groupByAvailability = (arr) => {
```
Comment (andrii-kuzmichov) [inline frontend/src/components/ScheduleDialog/ScheduleDialog.js:25]: move to utils

File: frontend/src/components/ScheduleDialog/ScheduleDialog.js [hunk 19]
```diff
@@ -49,138 +66,112 @@ const ScheduleDialog = (props) => {
...
-        getOptionLabel: (option) =>
-            option
-                ? option.available
-                    ? `${option.name} (${translation('common:available')})`
-                    : `${option.name} (${translation('common:unavailable')})`
-                : '',
+        getOptionLabel,
     };
 
+    if (sure && isLoading)
```
Comment (andrii-kuzmichov) [inline frontend/src/components/ScheduleDialog/ScheduleDialog.js:74]: `sure` what is it?

File: frontend/src/components/ScheduleDialog/ScheduleDialog.js [hunk 20]
```diff
@@ -49,138 +66,112 @@ const ScheduleDialog = (props) => {
...
+        return (
+            <div className="circular-progress-dialog">
+                <CircularProgress />
+            </div>
+        );
+
+    const surePositive = (
```
Comment (andrii-kuzmichov) [inline frontend/src/components/ScheduleDialog/ScheduleDialog.js:81]: the same Q

File: frontend/src/components/ScheduleDialog/ScheduleDialog.js [hunk 21]
```diff
@@ -49,138 +66,112 @@ const ScheduleDialog = (props) => {
...
+                >
+                    {translation('formElements:choose_button_title')}
+                </Button>
+                <Button className="dialog-button" variant="contained" onClick={() => onClose()}>
+                    {translation('formElements:cancel_button_title')}
+                </Button>
+            </div>
+        </>
+    );
+    const sureNegative = (
```
Comment (andrii-kuzmichov) [inline frontend/src/components/ScheduleDialog/ScheduleDialog.js:130]: the same

File: frontend/src/components/ScheduleItem/ScheduleItem.js [hunk 22]
```diff
@@ -3,39 +3,25 @@ import { MdDelete, MdEdit } from 'react-icons/md';
...
     const t = props.translation;
-    const { fStrLetterCapital } = props;
+    const { fStrLetterCapital, addition } = props;
 
     if (item) {
-        lesson = item.lesson;
-        itemId = item.id;
+        const { lesson: propsItemLesson, id: propsItemId } = item;
+        lesson = propsItemLesson;
+        itemId = propsItemId;
```
Comment (andrii-kuzmichov) [inline frontend/src/components/ScheduleItem/ScheduleItem.js:19]: why you made it?

File: frontend/src/components/ScheduleLessonsList/ScheduleLessonsList.js [hunk 23]
```diff
@@ -27,8 +27,8 @@ const ScheduleLessonsList = (props) => {
         if (group) selectGroupIdService(group.id);
     };
 
-    const groupFinderHandle = (groupId) => {
-        if (groupId) return groups.find((group) => group.id === groupId);
+    const groupFinderHandle = (groupIdProp) => {
+        if (groupIdProp) return groups.find((group) => group.id === groupIdProp);
```
Comment (andrii-kuzmichov) [inline frontend/src/components/ScheduleLessonsList/ScheduleLessonsList.js:31]: redundant renaming

File: frontend/src/components/SemesterCopyForm/SemesterCopyForm.js [hunk 24]
```diff
@@ -31,35 +29,34 @@ const SemesterCopyForm = (props) => {
...
+                            </option>
+                        ))}
+                    </Field>
+                </>
+            );
+        }
+        if (availableSemestersForCopy && availableSemestersForCopy.length === 1) {
+            return <p>{availableSemestersForCopy[0].description}</p>;
         }
+        return null;
```
Comment (andrii-kuzmichov) [inline frontend/src/components/SemesterCopyForm/SemesterCopyForm.js:60]: switch case

File: frontend/src/components/SemesterCopyForm/SemesterCopyForm.js [hunk 25]
```diff
@@ -31,35 +29,34 @@ const SemesterCopyForm = (props) => {
...
-                        </Field>
-                    </>
-                );
-            }
-            if (available_semesters_for_copy.length === 1) {
-                return <p>{available_semesters_for_copy[0].description}</p>;
-            }
+        if (availableSemestersForCopy && availableSemestersForCopy.length > 1) {
+            return (
+                <>
```
Comment (andrii-kuzmichov) [inline frontend/src/components/SemesterCopyForm/SemesterCopyForm.js:38]: drop `<>`

File: frontend/src/components/SemesterForm/SemesterForm.js [hunk 26]
```diff
@@ -22,8 +22,35 @@ import { getClearOrCancelTitle, setDisableButton } from '../../helper/disableCom
...
 const AddSemesterForm = (props) => {
+    const prepSetCheckedClasses = {};
+    const getToday = () => {
+        return new Date();
+    };
+    const getTomorrow = () => {
+        const tomorrow = new Date(getToday());
+        tomorrow.setDate(tomorrow.getDate() + 1);
+        return tomorrow;
+    };
```
Comment (andrii-kuzmichov) [inline frontend/src/components/SemesterForm/SemesterForm.js:36]: move to utils

File: frontend/src/components/SemesterForm/SemesterForm.js [hunk 27]
```diff
@@ -22,8 +22,35 @@ import { getClearOrCancelTitle, setDisableButton } from '../../helper/disableCom
 import { showAllGroupsService } from '../../services/groupService';
 import { MultiselectForGroups } from '../../helper/MultiselectForGroups';
 import { isObjectEmpty } from '../../helper/ObjectRevision';
+import { dateFormat } from '../../constants/formats';
 
 const AddSemesterForm = (props) => {
+    const prepSetCheckedClasses = {};
+    const getToday = () => {
+        return new Date();
+    };
```
Comment (andrii-kuzmichov) [inline frontend/src/components/SemesterForm/SemesterForm.js:31]: drop

File: frontend/src/components/SemesterForm/SemesterForm.js [hunk 28]
```diff
@@ -54,17 +83,18 @@ const AddSemesterForm = (props) => {
...
-        }
-    }, [semester.id]);
     const getGroupOptions = (groupOptions) => {
         return groupOptions.map((item) => {
             return { id: item.id, value: item.id, label: `${item.title}` };
         });
     };
+    useEffect(() => {
+        const { semesterGroups } = semester;
+        if (semesterGroups !== undefined && semesterGroups !== null) {
```
Comment (andrii-kuzmichov) [inline frontend/src/components/SemesterForm/SemesterForm.js:93]: `isNil` from lodash

File: frontend/src/components/SemesterForm/SemesterForm.js [hunk 29]
```diff
@@ -54,17 +83,18 @@ const AddSemesterForm = (props) => {
...
     const [openGroupDialog, setOpenGroupDialog] = useState(false);
-    useEffect(() => {
-        const { semester_groups } = semester;
-        if (semester_groups !== undefined && semester_groups !== null) {
-            setSelectedGroups(getGroupOptions(semester_groups));
-        }
-    }, [semester.id]);
     const getGroupOptions = (groupOptions) => {
         return groupOptions.map((item) => {
             return { id: item.id, value: item.id, label: `${item.title}` };
```
Comment (andrii-kuzmichov) [inline frontend/src/components/SemesterForm/SemesterForm.js:88]: drop `return` word

**Review by andrii-kuzmichov — COMMENTED** [review]
Please let self review before team review

**Review by andrii-kuzmichov — APPROVED** [review]

**Pull request merged**